pkg/signup/signup_service.go

```diff
@@ -240,25 +242,47 @@ func (s *ServiceImpl) GetSignup(userID string) (*Signup, error) {
 }
 
 // GetUserSignup is used to return the actual UserSignup resource instance, rather than the Signup DTO
-func (s *ServiceImpl) GetUserSignup(userID string) (*crtapi.UserSignup, error) {
+func (s *ServiceImpl) GetUserSignup(userID string) (*v1alpha1.UserSignup, error) {
 	// Retrieve UserSignup resource from the host cluster
 	userSignup, err := s.UserSignups.Get(userID)
 	if err != nil {
-		if errors.IsNotFound(err) {
-			return nil, nil
-		}
 		return nil, err
 	}
 
 	return userSignup, nil
 }
 
 // UpdateUserSignup is used to update the provided UserSignup resource, and returning the updated resource
-func (s *ServiceImpl) UpdateUserSignup(userSignup *crtapi.UserSignup) (*crtapi.UserSignup, error) {
+func (s *ServiceImpl) UpdateUserSignup(userSignup *v1alpha1.UserSignup) (*v1alpha1.UserSignup, error) {
 	userSignup, err := s.UserSignups.Update(userSignup)
 	if err != nil {
 		return nil, err
 	}
 
 	return userSignup, nil
 }
+
+// PhoneNumberAlreadyInUse checks if the phone number has been banned. If so, return
+// an internal server error. If not, check if a signup with a different userID
+// exists. If so, return an internal server error. Otherwise, return without error.
+func (s *ServiceImpl) PhoneNumberAlreadyInUse(userID, countryCode, phoneNumber string) error {
+	bannedUserList, err := s.BannedUsers.ListByHashedLabel(countryCode+phoneNumber, v1alpha1.BannedUserPhoneNumberHashLabelKey)
+	if err != nil {
+		return errors3.NewInternalError(err, "failed listing banned users")
+	}
+	if len(bannedUserList.Items) > 0 {
+		return errors3.NewForbiddenError("cannot re-register with phone number", "phone number already in use")
+	}
+
+	userSignupList, err := s.UserSignups.ListByHashedLabel(countryCode+phoneNumber, v1alpha1.UserSignupUserPhoneHashLabelKey)
+	if err != nil {
+		return errors3.NewInternalError(err, "failed listing userSignups")
+	}
+	for _, signup := range userSignupList.Items {
+		if signup.Name != userID {
+			return errors3.NewForbiddenError("cannot re-register with phone number", "phone number already in use")
+		}
+	}
+
+	return nil
+}
```

Comment:
Any tests for PhoneNumberAlreadyInUse()?

Reply:
added as part of: 956c1f0 and 7a21751